jbpm-event-emitters/jbpm-event-emitters-kafka/src/test/java/org/jbpm/event/emitters/kafka/KafkaEventEmitterTest.java

```diff
@@ -0,0 +1,183 @@
+            assertEquals("server-1", pi.get("compositeId"));
+            assertEquals("container-1", pi.get("containerId"));
+            assertEquals("process-1", pi.get("correlationKey"));
+            assertEquals(1, pi.get("id"));
+            assertEquals(0, pi.get("parentId"));
+            assertEquals("Test", pi.get("processId"));
+            assertEquals("pepe", pi.get("initiator"));
+            assertEquals(1, pi.get("state"));
+            assertEquals("a process", pi.get("processInstanceDescription"));
+            assertEquals("Test", pi.get("processName"));
+            assertEquals("1_0", pi.get("processVersion"));
+            assertTrue(pi.get("variables") instanceof Map);
+            assertTrue(((Map<?, ?>) pi.get("variables")).isEmpty());
+
+            record = producedEvents.get(1);
+            assertEquals("jbpm-tasks-events", record.topic());
+            Map<String, Object> taskEvent = mapper.readValue(record.value(), Map.class);
+            assertEquals("task", taskEvent.get("type"));
+            assertEquals("/process/Test/1", taskEvent.get("source"));
+            assertTrue(taskEvent.get("data") instanceof Map);
+            Map<String, Object> task = (Map<String, Object>) taskEvent.get("data");
+            assertEquals("Test", task.get("processId"));
+            assertEquals(1, task.get("processInstanceId"));
+            assertEquals("pepe", task.get("actualOwner"));
+
+            record = producedEvents.get(2);
+            assertEquals("customer-cases", record.topic());
+            Map<String, Object> caseEvent = mapper.readValue(record.value(), Map.class);
+            assertEquals("case", caseEvent.get("type"));
+            assertEquals("/process/Test/1", caseEvent.get("source"));
+            assertTrue(caseEvent.get("data") instanceof Map);
+            Map<String, Object> case1 = (Map<String, Object>) caseEvent.get("data");
+            assertEquals("Test", case1.get("caseDefinitionId"));
+            assertEquals(1, case1.get("id"));
+            assertEquals("pepe", case1.get("owner"));
+        } finally {
+            System.clearProperty("org.kie.jbpm.event.emitters.kafka.topic.cases");
+        }
+    }
+
+    @Test
+    public void testProducerFailure() throws IOException, ParseException {
+
+        InstanceView<ProcessInstance> piView = new InstanceView<ProcessInstance>() {
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public ProcessInstance getSource() {
+                return null;
+            }
+
+            @Override
+            public void copyFromSource() {
+
+            }
+        };
+        try (MockKafkaEventEmitter emitter = new MockKafkaEventEmitter()) {
+            emit(emitter, Collections.singletonList(piView));
+            List<ProducerRecord<String, byte[]>> producedEvents = emitter.producer.history();
+            assertEquals(0, producedEvents.size());
+        }
+    }
+
```

Comment:
Could you add any other rainy test for coverage? e.g. pass data null or empty or provoke an exception at send

Reply:
Yes, done